README.md
[![Build Status](https://travis-ci.org/servisys/ispconfig_setup.svg?branch=master)](https://travis-ci.org/servisys/ispconfig_setup)

# README #

[![PayPayl donate button](https://www.paypalobjects.com/it_IT/IT/i/btn/btn_donateCC_LG.gif)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=TB4Q3UJDC5JDJ "Help US support this project using Paypal")

# Newsletter #
Subscribe to our newsletter to receive information about new version of the script
The link is here: http://eepurl.com/cAzq95
We'll use only to inform you on new version of the script :)


# Version #
<b>v.3.0.6</b>

Debian 10 fix for Debian 10

<b>v.3.0.5</b>

Debian 10 changes file path, for root use :

	su -
	
Added support for Debian 10 Buster and php7.3
	service changed to systemctl for all service functions (start,stop,restart etc)
	few other minor changes to accommodate Debian 10

<b>v.3.0.4</b>

This is a system to automate the installation of ISPConfig 3 control Panel ( http://www.ispconfig.org/page/home.html ).

Tested on:
- Debian 10 Busty ([Servisys VPS](https://www.servisys.it/), VmWare Esxi, Amazon AWS, Virtualbox, OVH VPS, Hetzner, Digital Ocean)
- Debian 9 Stretch ([Servisys VPS](https://www.servisys.it/), VmWare Esxi, Amazon AWS, Virtualbox, OVH VPS, Hetzner, Digital Ocean)
- Debian 8 Jessie ([Servisys VPS](https://www.servisys.it/), VmWare Esxi, Amazon AWS, Virtualbox, OVH VPS, Hetzner, Digital Ocean)
- Debian 7 Wheezy ([Servisys VPS](https://www.servisys.it/), VmWare Esxi, Amazon AWS, Virtualbox, OVH VPS, Hetzner, Digital Ocean)
- Ubuntu 14.04 Trusty ([Servisys VPS](https://www.servisys.it/), VmWare Esxi, Amazon AWS, Virtualbox, OVH VPS, Hetzner, Digital Ocean)
- Ubuntu 15.10 Willy ([Servisys VPS](https://www.servisys.it/), VmWare Esxi, Amazon AWS, Virtualbox, OVH VPS, Hetzner, Digital Ocean)
- Ubuntu 16.04 Xenial Xerus ( [Servisys VPS](https://www.servisys.it/), VmWare Esxi, Amazon AWS, Virtualbox, OVH VPS, Hetzner, Digital Ocean)
- Ubuntu 18.04 Bionic Beaver ( [Servisys VPS](https://www.servisys.it/), VmWare Esxi, Amazon AWS, Virtualbox, OVH VPS, Hetzner, Digital Ocean)
- CentOS 7 ([Servisys VPS](https://www.servisys.it/), Vitualbox)
- Raspbian
- ISPConfig 3.*

### What is this repository for? ###

This repository contains some scripts for the automation of installation of ISPConfig 3 control panel.

Before starting, be sure to follow one of these guides to install a supported Linux distribution:

- Debian 7: https://www.howtoforge.com/tutorial/debian-7-wheezy-minimal-server/
- Debian 8: https://www.howtoforge.com/tutorial/debian-8-jessie-minimal-server/
- Debian 9: https://www.howtoforge.com/tutorial/debian-minimal-server/
- Debain 10: https://www.howtoforge.com/tutorial/debian-10-buster-minimal-server/
- Ubuntu 14.10: https://www.howtoforge.com/tutorial/ubuntu-14.10-utopic-unicorn-server
- Ubuntu 15.10: https://www.howtoforge.com/tutorial/ubuntu-15.10-wily-werewolf-minimal-server/
- Ubuntu 16.04: https://www.howtoforge.com/tutorial/ubuntu-16.04-xenial-xerus-minimal-server/
- Ubuntu 16.10: https://www.howtoforge.com/tutorial/ubuntu-16.10-yakkety-yak-minimal-server/
- Ubuntu 17.10: https://www.howtoforge.com/tutorial/ubuntu-minimal-server-install/
- Ubuntu 18.04: https://www.howtoforge.com/tutorial/ubuntu-lts-minimal-server/
- CentOS 7: https://www.howtoforge.com/tutorial/centos-7-minimal-server/

#### Supported Software and Linux distributions
<table cellpadding="0" cellspacing="0">
	<tr>
		<td rowspan="2"><strong>Component</strong></td>
		<td rowspan="2"><strong>Software</strong></td>
		<td colspan="4"><strong>Debian/Raspbian</strong></td>
		<td colspan="6"><strong>Ubuntu</strong></td>
		<td><strong>CentOS</strong></td>
		<td colspan="2"><strong>openSUSE Leap</strong></td>
		<td><strong>Fedora</strong></td>
	</tr>
	<tr>
		<td>7</td>
		<td>8</td>
        <td>9</td>
        <td>10</td>
		<td>14.04</td>
		<td>15.10</td>
		<td>16.04</td>
		<td>16.10</td>
		<td>17.10</td>
		<td>18.04</td>
		<td>7</td>
		<td>42.1-3</td>
		<td>15.0</td>
		<td>22-28</td>
	</tr>
	<tr>
		<td rowspan="2">Web: HTTP</td>
		<td>Apache</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>nginx</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>Mail: SMTP</td>
		<td>Postfix</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td rowspan="2">Mail: POP3/IMAP</td>
		<td>Courier</td>
		<td>✔</td>
		<td></td>
        <td></td>
        <td></td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>Dovecot</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>FTP</td>
		<td>Pure-FTPd</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td rowspan="3">DNS</td>
		<td>Bind</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>PowerDNS</td>
		<td></td>
		<td></td>
        <td></td>
        <td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>MyDNS</td>
		<td></td>
		<td></td>
        <td></td>
        <td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td rowspan="2">Database</td>
		<td>MySQL</td>
		<td>✔</td>
		<td>✔</td>
        <td></td>
        <td></td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>MariaDB</td>
		<td></td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td rowspan="2">Webmail client</td>
		<td>Roundcube</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td></td>
		<td></td>
		<td>✔*</td>
		<td></td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>SquirrelMail</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td></td>
		<td>✔</td>
		<td>✔</td>
		<td>✔*</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>Chat: XMPP</td>
		<td>Metronome</td>
		<td></td>
		<td></td>
        <td></td>
        <td></td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔*</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>Mailing lists</td>
		<td>Mailman</td>
		<td></td>
		<td></td>
        <td></td>
        <td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td rowspan="2">Antivirus</td>
		<td>Amavisd</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>ClamAV</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>Spam filtering</td>
		<td>SpamAssassin</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>Greylisting</td>
		<td>Postgrey</td>
		<td></td>
		<td>✔</td>
        <td>✔</td>
        <td></td>
		<td></td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td rowspan="2">Mail signing</td>
		<td>OpenDKIM</td>
		<td></td>
		<td>✔</td>
        <td>✔</td>
        <td></td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>OpenDMARC</td>
		<td></td>
		<td>✔</td>
        <td>✔</td>
        <td></td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>Firewall</td>
		<td>UFW</td>
		<td></td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td></td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>Intrusion protection</td>
		<td>Fail2Ban</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>Rootkit detection</td>
		<td>rkhunter</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td rowspan="2">Statistics</td>
		<td>Webalizer</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>AWStats</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td></td>
		<td>Quota</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔*</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>Let's Encrypt</td>
		<td>Certbot/letsencrypt</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td></td>
		<td>Jailkit</td>
		<td>✔</td>
		<td>✔</td>
        <td>✔</td>
        <td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td></td>
		<td>HHVM</td>
		<td></td>
		<td>✔^</td>
        <td>✔^</td>
        <td></td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td>✔</td>
		<td></td>
		<td>✔</td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
	<tr>
		<td>MultiServer</td>
		<td></td>
		<td></td>
        <td>✔</td>
        <td>✔</td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
		<td></td>
	</tr>
</table>
* Not yet enabled.
^ Not yet enabled on Raspbian.


You can choose during install:
- Apache or nginx
- Dovecot or Courier
- Quota
- Jailkit
- SquirrelMail or Roundcube
- ISPConfig 3 Standard / Expert mode
- ISPConfig 3 Multiserver Setup (* Debian 8 only for now)

### How do I get set up? ###

Follow one of the above guides to install a fresh copy of a supported Linux distribution.

Run the following command as root user:

* Debian/Raspbian 7, 8, 9 and 10 and Ubuntu 14.04, 15.10, 16.04, 16.10 and 18.04

```shell
cd /tmp; wget -O installer.tgz "https://github.com/servisys/ispconfig_setup/tarball/master"; tar zxvf installer.tgz; cd *ispconfig*; bash install.sh
```
* CentOS 7

```shell
cd /tmp; sudo yum install wget unzip net-tools; wget -O installer.tgz "https://github.com/servisys/ispconfig_setup/tarball/master"; tar zxvf installer.tgz; cd *ispconfig*; sudo install.sh
```

CentOS 7 is in a very early stage, we got to test a bit, any help will be appreciated. 
Some features are missing for now, only implemented Apache and Dovecot, no webmail.

If `wget` fails, try adding the `--no-check-certificate` parameter.

Follow the instructions on the screen

### Who had contributed to this work? ###

* The scripts and instructions have been produced by Matteo Temporini ( <temporini.matteo@gmail.com> )
* Special thanks to Travis CI for adding support to Raspberry and a big number of Bugs( https://github.com/tdulcet )
* Special thanks to Torsten Widmann for contribution to the code
* Special thanks to Michiel Kamphuis ( http://www.ensync.it/ ) for contribution to Multiserver Setup integration
* Special thanks to Bartłomiej Gajda ( http://webard.me/ ) for the bug fixes to multiserver setup and beta installation
* The code is based on the "Automatic Debian System Installation for ISPConfig 3" of Author: Mark Stunnenberg <mark@e-rave.nl>
* Howtoforge community https://www.howtoforge.com/community/


CHANGELOG.txt
v.3.0.6
- Added fix for Debian 10
- Removed support for multiserver setup in Debian 9 and Debian 10

v.3.0.5
- Added support for debain10 and php7.3

v.3.0.4
- Added Raspbian support
- Bug fix

v.3.0.3.1
- Added small fix for ubuntu 18.04

v.3.0.3
- Added support for Ubuntu 18.04

v.3.0.2
- Added support for nginx on Debian 9
- Phpmyadmin accessible on http:/IPADDRESS:8081/phpmyadmin
- Roundcube accessible at https://IPADDRESS/roundcube/

v.3.0.1
- Added support for both PHP7 and PHP 5.6 in Debian 9

v.3.0.0
- Added support for Debian 9 - PHP 7 - Apache and Roundcube webmail - MariaDB as SQL

v.2.2.6
- Fix for Centos 7

v.2.2.5
- Fix for Centos 7
- Small fix for debian multiserver setup and version installation

v.2.2.4
- various fix for Ubuntu 16.04

v.2.2.3
- Lot of fix for Ubuntu 16.04
- fix for asking password in ispconfig autoinstall.ini parameter

v.2.2.2
- Fixed php7 install on ubuntu 16.04
- Fixed php7-fpm with nginx

v.2.2.1
- Added support for let's encrypt for ubuntu 14.04, 15.10, 16.04, debian 7 and debian 8, for both apache2 and nginx

v.2.2
- Added support for ubuntu 16.04
- working with ispconfig 3.1
- gradually removing support for old versions of ispconfig

v.2.1.1
- Lot of bugfix ( thanks to webard )

v.2.1.0
- Adding support for ISPConfig 3.1 beta (Only for Debian 8 for now)

v.2.0.4
- Fixed issue Ubuntu 15.10/14.04 - [: ==: unary operator expected #92 

v.2.0.3
- Aded support for Roundcube Webmail on Debian 8 through Backports

v.2.0.2
- Small fix after merge 2.0.1

v.2.0.1
- Fixed Issue 87 (Squirrelmail related)

v.2.0.0
- Added Multiserver setup configuration on Debian 8 (use carfeully, and only if know what are you doing)
- Improvment of log during install (the whole process is logged now)

v.1.0.16
- Added support for Support ubuntu 15.10

v.1.0.15
- Added support for ubuntu 14.04 LTS

v.1.0.14
- Starting developing for centos 7
- Added implementation for Apache con Centos 7
- Added implementation for Postfix con Centos 7
- Added implementation for MySQL con Centos 7
- Added implementation for Amavis con Centos 7

Known missing part
- No webmail installed
- no courier support
- no nginx support
- no quota support
- use with caution for now and don't use on production systems.

v.1.0.13
- Fix phpmyadmin install process hangs
- Added warning for roundcube install to modify config for rounducbe remote user and pass 
- Disable fstab edit for quota in OpenVZ environment
- Removed systemd install during setup, causing problem after installation on some Debian 7 systems

v.1.0.12
- Added operation system detect based on /etc/os-relase
- Added stop installation check if no suported os detected
- Fix on nginx/apache restart

v.1.0.11
- Added optional Install for phpMyAdmin & XCache (Debian 7/Debian 8)

v.1.0.10
- Adding support for ISPConfig expert mode setup
- Fix problem in debian 7 for install MySQL

v.1.0.9
- Added support for MariaDB 
- Added fix for Nginx on Jessie
- Smallfix installazion

v.1.0.8
- Added support for debian Jessie
- Added Module for all
- Fix minor issue
- Remove Roundcube support for debian Jessie installation

v.1.0.6
- Starting modular system for better managment and integration
- Added more presintall check
- Added check for different distribution (Ubuntu, Debian Wheezy end Debian Jessie)
- Little fix and improvments
- Added check for sendmail installation and removal
- Added rkhunter
- Added jailkit version through simple editable variable
- Changed dovect as default mta
- Skip quota adding for tmpfs fylesystem

v.1.0.5
- Adding support for nGinx

v.1.0.4
- Adding log errors to /var/log/ispconfig_setup.log 
- Added a2enmod fcgid, who caused panel not to work
- Added DKIM support in amavisd-new (disabled by default)
- Fixed issue in /etc/php5/conf.d/ming.ini changin comment from # to ;
- Added some color and clear screen after installation is started

v.1.0.2
- Added support for inserit SSL data only one time, and then used all the time SSL is needed
- Added possibility to insert blank information in SSL filed for quick setup
- ISPconfig install process, completelly automated, no intereaction required
- Fixed update of freshclam after install and restart demon
- Added InstallFix pprocedure to do postinstall necessary fix
- Fixed roundcube user configuration 

v.1.0.1
- check on apt repository needed for ISPConfig
- No need of external files

v.1.0.0
- Very first version of the script


functions/check_linux.sh
#---------------------------------------------------------------------
# Function: CheckLinux
#	Check Installed Linux Version
#---------------------------------------------------------------------

# Adapted from: https://github.com/mail-in-a-box/mailinabox/blob/master/setup/functions.sh#L1
function hide_output {
	OUTPUT=$(mktemp)

	# <&-
	"$@" &> "$OUTPUT"

	E=$?
	if [[ $E -ne 0 ]]; then
		echo -e "\nError: The following command failed: $*\n"
		echo "-----------------------------------------"
		cat "$OUTPUT"
		echo "-----------------------------------------"
		echo
		rm -f "$OUTPUT"
		exit $E
	fi

	rm -f "$OUTPUT"
}

# Adapted from: https://github.com/mail-in-a-box/mailinabox/blob/master/setup/functions.sh#L41
function apt_install {
	hide_output apt-get -y install "$@"
}

function apt_remove {
	hide_output apt-get -y remove "$@"
}

function yum_install {
	hide_output yum -y install "$@"
}

function yum_remove {
	hide_output yum -y erase "$@"
}

function zypper_install {
	hide_output zypper in -y "$@"
}

function zypper_remove {
	hide_output zypper rm -y "$@"
}

function dnf_install {
	hide_output dnf -y install "$@"
}

function dnf_remove {
	hide_output dnf -y erase "$@"
}

CheckLinux() {

	#Extract information on system
	. /etc/os-release

	# Set DISTRO variable to null
	DISTRO=''
	
	if echo "$ID" | grep -iq "debian"; then

		#---------------------------------------------------------------------
		#	Debian 7 Wheezy
		#---------------------------------------------------------------------
		
		if echo "$VERSION_ID" | grep -iq "7"; then
			DISTRO=debian7

		#---------------------------------------------------------------------
		#	Debian 8 Jessie
		#---------------------------------------------------------------------
		
		elif echo "$VERSION_ID" | grep -iq "8"; then
			DISTRO=debian8

		#---------------------------------------------------------------------
		#	Debian 9 Stretch
		#---------------------------------------------------------------------
		
		elif echo "$VERSION_ID" | grep -iq "9"; then
			DISTRO=debian9

		#---------------------------------------------------------------------
		#	Debian 10 Buster
		#---------------------------------------------------------------------
		
		elif echo "$VERSION_ID" | grep -iq "10"; then
			DISTRO=debian10
		fi
	
	elif echo "$ID" | grep -iq "raspbian"; then
	
		#---------------------------------------------------------------------
		#	Raspbian 7 Wheezy
		#---------------------------------------------------------------------
		
		if echo "$VERSION_ID" | grep -iq "7"; then
			DISTRO=debian7

		#---------------------------------------------------------------------
		#	Raspbian 8 Jessie
		#---------------------------------------------------------------------
		
		elif echo "$VERSION_ID" | grep -iq "8"; then
			DISTRO=debian8

		#---------------------------------------------------------------------
		#	Raspbian 9 Stretch
		#---------------------------------------------------------------------
		
		elif echo "$VERSION_ID" | grep -iq "9"; then
			DISTRO=debian9
		fi
	
	elif echo "$ID" | grep -iq "ubuntu"; then
	
		#---------------------------------------------------------------------
		#	Ubuntu 14.04
		#---------------------------------------------------------------------
		
		if echo "$VERSION_ID" | grep -iq "14.04"; then
			DISTRO=ubuntu-14.04

		#---------------------------------------------------------------------
		#	Ubuntu 15.10
		#---------------------------------------------------------------------

		elif echo "$VERSION_ID" | grep -iq "15.10"; then
			DISTRO=ubuntu-15.10
		
		#---------------------------------------------------------------------
		#	Ubuntu 16.04
		#---------------------------------------------------------------------

		elif echo "$VERSION_ID" | grep -iq "16.04"; then
			DISTRO=ubuntu-16.04
		
		#---------------------------------------------------------------------
		#	Ubuntu 16.10
		#---------------------------------------------------------------------

		elif echo "$VERSION_ID" | grep -iq "16.10"; then
			DISTRO=ubuntu-16.10
		
		#---------------------------------------------------------------------
		#	Ubuntu 17.10
		#---------------------------------------------------------------------

		elif echo "$VERSION_ID" | grep -iq "17.10"; then
			DISTRO='' #ubuntu-17.10
		
		#---------------------------------------------------------------------
		#	Ubuntu 18.04
		#---------------------------------------------------------------------

		elif echo "$VERSION_ID" | grep -iq "18.04"; then
			DISTRO=ubuntu-18.04
		fi
	
	elif echo "$ID" | grep -iq "centos"; then
	
		#---------------------------------------------------------------------
		#	CentOS
		#---------------------------------------------------------------------

		if echo "$VERSION_ID" | grep -iq "7"; then
			DISTRO=centos7
		fi
	
	elif echo "$ID" | grep -iq "opensuse"; then
	
		#---------------------------------------------------------------------
		#	openSUSE Leap 42.1
		#---------------------------------------------------------------------

		if echo "$VERSION_ID" | grep -iq "42.1"; then
			DISTRO='' #opensuse-42.1
		
		#---------------------------------------------------------------------
		#	openSUSE Leap 42.2
		#---------------------------------------------------------------------

		elif echo "$VERSION_ID" | grep -iq "42.2"; then
			DISTRO='' #opensuse-42.2
		
		#---------------------------------------------------------------------
		#	openSUSE Leap 42.3
		#---------------------------------------------------------------------

		elif echo "$VERSION_ID" | grep -iq "42.3"; then
			DISTRO='' #opensuse-42.3
		
		#---------------------------------------------------------------------
		#	openSUSE Leap 15.0
		#---------------------------------------------------------------------

		elif echo "$VERSION_ID" | grep -iq "15.0"; then
			DISTRO='' #opensuse-15.0
		fi
	
	elif echo "$ID" | grep -iq "fedora"; then
	
		#---------------------------------------------------------------------
		#	Fedora 22 - 28
		#---------------------------------------------------------------------

		RE='^[0-9]+$'
		if [[ "$VERSION_ID" =~ $RE && "$VERSION_ID" -ge 22 && "$VERSION_ID" -le 28 ]]; then
			DISTRO='' #fedora
		fi
	fi

}

